### DISCUSSION TITLE
impute missing column values with mean of subset of values

### DISCUSSION THREAD

**Discussion by apintabir** [uwdata/arquero#316] [Q&A]
Hi, I recently started using arquero and I've mostly been able to get up to speed with the (very helpful) documentation. However, I can't wrap my head around how to do the following:  
 
I have a table with a column that has a bunch of missing values, e.g.:
{ b: [11.91, NA, 27.25, 33.11, NA, 480.30, NA, 4.17, 17.89, 14.0, 7, 2.19, 1.22, 8.43, 3.10] }

I want to impute the missing values, but only with the mean of the bottom 25% of the values in the column (i.e. with 2.17, the average of 1.22, 2.19, 3.10). Is there a way to do this without having to escape outside the table and store the mean value as a variable? 

Thanks!

**Comment by markwhiting** [comment]
I'm sure you have found a solution by now, but if not, the [table.**impute**](https://uwdata.github.io/arquero/api/verbs#impute) verb can do this for you. There are some example of it in use in the [An Illustrated Guide to Arquero Verbs](https://observablehq.com/@uwdata/an-illustrated-guide-to-arquero-verbs#impute) on Observable.
